streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -513,22 +507,31 @@ class StreamingContext private[streaming] (
   }
 
   /**
+   * Return the current state of the context.
+   */
+  def getState(): StreamingContextState = {
+    state
+  }
+
+  /**
    * Start the execution of the streams.
    *
    * @throws SparkException if the context has already been started or stopped.
    */
   def start(): Unit = synchronized {
-    if (state == Started) {
-      throw new SparkException("StreamingContext has already been started")
-    }
-    if (state == Stopped) {
-      throw new SparkException("StreamingContext has already been stopped")
+    state match {
+      case INITIALIZED =>
+        // good to start
+      case STARTED =>
```

Comment:
Is there a problem if the start() call gets called again? Why not simply be a no-op if the context is started? It might be user-code bug/race condition, but it really does indicate an error state, does it?